meta-facebook/yv35-rf/src/ipmi/plat_ipmi.c

```diff
@@ -43,3 +47,49 @@ void OEM_1S_GET_BOARD_ID(ipmi_msg *msg)
 	msg->completion_code = CC_SUCCESS;
 	return;
 }
+
+void OEM_1S_GET_FW_VERSION(ipmi_msg *msg)
+{
+	CHECK_NULL_ARG(msg);
+	if (msg->data_len != 1) {
+		msg->completion_code = CC_INVALID_LENGTH;
+		return;
+	}
+	bool ret = false;
+	uint8_t component = msg->data[0];
+	mctp *mctp_inst = NULL;
+	mctp_ext_params ext_params = { 0 };
+	uint8_t read_len = 0;
+	uint8_t resp_buf[GET_FW_INFO_REVISION_LEN] = { 0 };
+
+	switch (component) {
+	case RF_COMPNT_BIC:
+		msg->data[0] = BIC_FW_YEAR_MSB;
+		msg->data[1] = BIC_FW_YEAR_LSB;
+		msg->data[2] = BIC_FW_WEEK;
+		msg->data[3] = BIC_FW_VER;
+		msg->data[4] = BIC_FW_platform_0;
+		msg->data[5] = BIC_FW_platform_1;
+		msg->data[6] = BIC_FW_platform_2;
+		msg->data_len = 7;
+		msg->completion_code = CC_SUCCESS;
+		break;
+	case RF_COMPNT_CXL:
+		if (get_mctp_info_by_eid(CXL_EID, &mctp_inst, &ext_params) == false) {
+			msg->completion_code = CC_UNSPECIFIED_ERROR;
+			return;
+		}
+		CHECK_NULL_ARG(mctp_inst);
+		ret = cci_get_chip_fw_version(mctp_inst, ext_params, resp_buf, &read_len);
+		msg->data[0] = component;
```

Comment:
Is there a need to check the ret to ensure the chip_fw_version is good before assignment?

Reply:
I have fixed this issue in the #cb065f commit.
Many thanks.